bitgouel-api/src/main/kotlin/team/msg/domain/lecture/service/LectureServiceImpl.kt

```diff
@@ -500,43 +523,51 @@ class LectureServiceImpl(
      * @param 이수 상태를 변경할 강의 id와 학생 id, 변결될 강의 여부
      */
     @Transactional(rollbackFor = [Exception::class])
-    override fun updateLectureCompleteStatus(id: UUID, studentId: UUID, isComplete: Boolean) {
+    override fun updateLectureCompleteStatus(id: UUID, studentIds: List<UUID>) {
         val user = userUtil.queryCurrentUser()
+        val students = studentRepository.findByIdIn(studentIds)
+        val lecture = lectureRepository findById id
 
         when(user.authority) {
             Authority.ROLE_TEACHER -> {
                 val teacher = teacherRepository findByUser user
-                val student = studentRepository findById studentId
 
-                if(teacher.club != student.club)
+                if(students.any { teacher.club != it.club })
                     throw ForbiddenSignedUpLectureException("학생의 이수 여부를 변경할 권한이 없습니다. info : [ userId = ${user.id} ]")
+
             }
             Authority.ROLE_BBOZZAK -> {
                 val bbozzak = bbozzakRepository findByUser user
-                val student = studentRepository findById studentId
 
-                if(bbozzak.club != student.club)
+                if(students.any { bbozzak.club != it.club })
                     throw ForbiddenSignedUpLectureException("학생의 이수 여부를 변경할 권한이 없습니다. info : [ userId = ${user.id} ]")
             }
             Authority.ROLE_ADMIN -> { }
             else -> {
-                val lecture = lectureRepository findById id
                 if (lecture.user != user)
                     throw ForbiddenSignedUpLectureException("학생의 이수 여부를 변경할 권한이 없습니다. info : [ userId = ${user.id} ]")
             }
         }
 
-        val registeredLecture = registeredLectureRepository.findByLectureIdAndStudentId(id, studentId)
-            ?: throw UnSignedUpLectureException("학생의 강의 신청 기록을 찾을 수 없습니다. info : [ lectureId = $id, studentId = $studentId ]")
-        
-        val updatedRegisteredLecture = RegisteredLecture(
-            id = registeredLecture.id,
-            student = registeredLecture.student,
-            lecture = registeredLecture.lecture,
-            isComplete = isComplete
-        )
+        val updatedRegisteredLectures = students.map { student ->
+            val registeredLecture = registeredLectureRepository.findByStudentAndLecture(student, lecture)
+                ?: throw UnSignedUpLectureException("학생의 강의 신청 기록을 찾을 수 없습니다. info : [ lectureId = $id, studentId = ${student.id} ]")
+
+            val completeStatus = when(student.grade) {
+                1 -> CompleteStatus.COMPLETED_IN_1RD
+                2 -> CompleteStatus.COMPLETED_IN_2RD
+                else -> CompleteStatus.COMPLETED_IN_3RD
```

Comment:
```suggestion
                1 -> CompleteStatus.COMPLETED_IN_1RD
                2 -> CompleteStatus.COMPLETED_IN_2RD
                3 -> CompleteStatus.COMPLETED_IN_3RD
                else -> throw ~~
```

위처럼 바꿔주는 것이 else에 대한 처리로 더 올바른 것 같습니다!

Reply:
[6bb7997](https://github.com/School-of-Company/Bitgouel-Server/pull/522/commits/6bb7997cb8ffc5c18c57b3c2c897603521ce8ef7) 변경해줬어요!